build.py

```diff
@@ -23,4 +28,98 @@ def decrypt():
+          print(json.dumps(remote_states, indent=1))
+        elif type(remote_states) == dict:
+          print(json.dumps(remote_states, indent=1))
+        else:
+          if quiet == 'False': print('No dependency detected.')
+    except Exception as e:
+        print('Error processing deps:',e)
+
+def _layer_dependency(directory):
+
+    config_files = ['config.tf']
+    config_file_contents = []
+
+    # #########################################
+    # get the config file contents
+    # #########################################
+    try:
+      for config_file in config_files:
+          with open(f"{directory}/{config_file}", 'r') as f:
+              config_file_contents.append(hcl2.load(f))
+    except Exception as e:
+        raise Exception(f"Can not open config files: {e}")
+
+    remote_states = {}
+
+    if len(config_file_contents)>0:
+      # #########################################
+      # remote states
+      # #########################################
+      # they have this shape
+      # data "terraform_remote_state" "eks-vpc" {
+      #   backend = "s3"
+      #   config = {
+      #     region  = var.region
+      #     profile = var.profile
+      #     bucket  = var.bucket
+      #     key     = "apps-prd/k8s-eks/network/terraform.tfstate"
+      #   }
+      # }
+      remote_state_resource_name = 'terraform_remote_state'
+      for data in config_file_contents:
+              if 'data' in data.keys():
+                  for d in data['data']:
+                      #print('--#',d)
+                      this_key = list(d.keys())[0]
+                      if this_key == remote_state_resource_name:
+                          #print('-->',d[this_key])
+                          this_name = list(d[this_key].keys())[0]
+                          #print('name',this_name)
+                          if 'config' in d[this_key][this_name] and 'key' in d[this_key][this_name]['config']:
+                              #print('   ---= dep:', d[this_key][this_name]['config']['key'])
+                              this_tfstate = d[this_key][this_name]['config']['key'].split('/')
+                              account = this_tfstate[0]
+                              layer = '/'.join(this_tfstate[1:-1])
+                              remote_states[this_name] = {'remote_state_name': this_name, 'account': account, 'layer': layer, 'key': d[this_key][this_name]['config']['key'], 'usage': {'used': False, 'files': []}}
+
+      # #########################################
+      # Where data is being used
+      # #########################################
+      remote_state_usage_regex = 'data\.terraform_remote_state\.([a-z_\-0-9]+)\..+'
+      files = glob.glob(f"{directory}/*tf")
+      for this_file in files:
+          with open(this_file, 'r') as f:
+              this_file_data = f.readlines()
+              for line in this_file_data:
+                  match_object = re.search(remote_state_usage_regex, line)
```

Comment:
```suggestion
                  if match_object := re.search(remote_state_usage_regex, line)
```
You could use the walrus operator here :slightly_smiling_face:

Reply:
also a great suggestion!